File: frontend/src/lib/config/evidence.ts
export enum Evidence {
	INFERRED = 'inferred',
	OBSERVED = 'observed',
	CORROBORATED = 'corroborated',
	PROVEN = 'proven'
}

export const EVIDENCE_ORDER: string[] = [
	Evidence.INFERRED,
	Evidence.OBSERVED,
	Evidence.CORROBORATED,
	Evidence.PROVEN
];

export const EVIDENCE_LABELS: Record<string, string> = {
	[Evidence.INFERRED]: 'Inferred',
	[Evidence.OBSERVED]: 'Observed',
	[Evidence.CORROBORATED]: 'Corroborated',
	[Evidence.PROVEN]: 'Proven'
};

export const EVIDENCE_HELP: Record<string, string> = {
	[Evidence.INFERRED]: 'The reported version falls inside the affected range.',
	[Evidence.OBSERVED]: 'A check matched the response it received.',
	[Evidence.CORROBORATED]: 'Two independent signals agree at the same location.',
	[Evidence.PROVEN]: 'The asset produced an out-of-band interaction.'
};

export function evidenceRank(value: string | null | undefined): number {
	return EVIDENCE_ORDER.indexOf(value ?? '');
}

export function evidenceLabel(value: string | null | undefined): string {
	return EVIDENCE_LABELS[value ?? ''] ?? '';
}

export function evidenceToken(value: string): string {
	return `evidence:${value}`;
}

File: shared/definitions/evidence.py
"""The evidence ladder: how a finding or inference was established."""

from __future__ import annotations

from enum import StrEnum


class Evidence(StrEnum):
    INFERRED = "inferred"
    OBSERVED = "observed"
    CORROBORATED = "corroborated"
    PROVEN = "proven"


EVIDENCE_ORDER: tuple[str, ...] = tuple(e.value for e in Evidence)

EVIDENCE_RANK: dict[str, int] = {value: i for i, value in enumerate(EVIDENCE_ORDER)}

EVIDENCE_LABELS: dict[str, str] = {
    Evidence.INFERRED.value: "Inferred",
    Evidence.OBSERVED.value: "Observed",
    Evidence.CORROBORATED.value: "Corroborated",
    Evidence.PROVEN.value: "Proven",
}

EVIDENCE_HELP: dict[str, str] = {
    Evidence.INFERRED.value: "The reported version falls inside the affected range.",
    Evidence.OBSERVED.value: "A check matched the response it received.",
    Evidence.CORROBORATED.value: "Two independent signals agree at the same location.",
    Evidence.PROVEN.value: "The asset produced an out-of-band interaction.",
}


def evidence_rank(value: str | None) -> int:
    return EVIDENCE_RANK.get(value or "", -1)


def evidence_label(value: str | None) -> str:
    return EVIDENCE_LABELS.get(value or "", "")
